### PR TITLE
[android_intent] componentName must be provided before resolveActivity is called

### PR DESCRIPTION
**Pull request by csobrinho**
If a componentName is provided, then the resolveActivity will fail since the componentName is only provided afterwards. This breaks intents of common actions like LAUNCHER that should point to a specific componentName.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by csobrinho** [comment]
Something is still missing. Please hold on the merge.

File: packages/android_intent/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.3.5
```
[the remark this answers is not in the document]
Comment (csobrinho) [reply packages/android_intent/CHANGELOG.md]: Done.

**Comment by csobrinho** [comment]
> Something is still missing. Please hold on the merge.

Got it. The map was using "component" instead of "componentName"

**Comment by csobrinho** [comment]
> If you don't have time to keep working on this let us know and somebody on our team will be happy to shepherd it forward from here.

Hi mklim@, I tried to run the robolectric tests but the CONTRIBUTING docs only explain how to run the instrumentation tests. Also when I run:

`~/src/flutter-plugins/packages/android_intent/android$ gradlew test`

I always get a bunch of undefined imports like:

```
> Task :compileDebugJavaWithJavac FAILED
src/flutter-plugins/packages/android_intent/android/src/main/java/io/flutter/plugins/androidintent/AndroidIntentPlugin.java:4: error: package io.flutter.embedding.engine.plugins does not exist
import io.flutter.embedding.engine.plugins.FlutterPlugin;
                                          ^
src/flutter-plugins/packages/android_intent/android/src/main/java/io/flutter/plugins/androidintent/AndroidIntentPlugin.java:5: error: package io.flutter.embedding.engine.plugins.activity does not exist
import io.flutter.embedding.engine.plugins.activity.ActivityAware;
                                                   ^
```

Do you mind merging this and then add the test in a follow up CL? We are currently blocked on this.

Also, might be a good idea to add to the [CONTRIBUTING.md](https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md) how to run these android unit tests.

Thanks

**Comment by mklim** [comment]
@csobrinho sorry about the missing documentation. Looking at that it's not even necessarily right for the tests it is describing, that should be updated. Added a note to flutter/flutter#35747. I've just been running these via Android Studio personally. They can also be run this way from the command line:

```
$ cd example
$ flutter build apk
$ cd android
$ ./gradlew test
```
That `flutter build apk` step is necessary to pull some Flutter specific dependencies in place first, and likely the cause of the random compile failures you were seeing. Sorry about that.

> Do you mind merging this and then add the test in a follow up CL? We are currently blocked on this.

Sorry to hear you're blocked. All PRs need to be tested unless given a personal exemption from Hixie (4 on [Tree Hygiene](https://github.com/flutter/flutter/wiki/Tree-hygiene)). Like I mentioned before I'm good to take over this PR and shepherd it forward. I'll add a test myself to help this land.

**Comment by mklim** [comment]
I pushed a test, will land on green.

**Comment by csobrinho** [comment]
Thanks!!

**Pull request merged**